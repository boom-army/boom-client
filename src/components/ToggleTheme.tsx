import React, { useState, useContext } from "react";
import { ThemeContext } from "../contexts/theme";
import { lightTheme, darkTheme, ThemeVars } from "../styles/themes";
import { ThemeIcon } from "./Icons";
import { styled } from "@mui/material/styles";

export const Wrapper = styled("div")({
  display: "flex",
  alignItems: "baseline",
  marginLeft: "0.7rem",
  marginBottom: "1rem",
  cursor: "pointer",
  p: {
    marginLeft: "0.4rem",
  },
});

const ToggleTheme = () => {
  const localSt = localStorage.getItem("theme");
  const [currentTheme, setCurrentTheme] = useState(localSt ? localSt : "dark");

  const { theme, setTheme } = useContext(ThemeContext);

  const toggleTheme = () => {
    if (currentTheme === "dark") {
      setTheme((theme: ThemeVars) => ({
        ...theme,
        ...lightTheme,
      }));
      localStorage.setItem("theme", "light");
      setCurrentTheme("light");
    } else {
      setTheme((theme: ThemeVars) => ({
        ...theme,
        ...darkTheme,
      }));
      localStorage.setItem("theme", "dark");
      setCurrentTheme("dark");
    }
  };

  return (
    <Wrapper onClick={toggleTheme}>
      <ThemeIcon sm="true" color={theme.accentColor} />
      <p>Theme</p>
    </Wrapper>
  );
};

export default ToggleTheme;
