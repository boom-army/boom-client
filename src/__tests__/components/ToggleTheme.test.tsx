import React from "react";
import ToggleTheme from "../../components/ToggleTheme";
import { act, fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("<ToggleTheme/> component :", () => {
  test("display <ToggleTheme/> component ", async () => {
    let rendered = render(<ToggleTheme />, { wrapper: MemoryRouter });
    expect(rendered).toMatchSnapshot();
  });
});

test("Assertion testing of <ToggleTheme/>  component", () => {
  render(<ToggleTheme />, { wrapper: MemoryRouter });
  expect(screen.getByText("Theme")).toBeInTheDocument();
});
