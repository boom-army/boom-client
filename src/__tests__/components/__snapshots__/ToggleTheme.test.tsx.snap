// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleTheme/> component : display <ToggleTheme/> component  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .css-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  margin-left: 0.7rem;
  margin-bottom: 1rem;
  cursor: pointer;
}

.css-0 p {
  margin-left: 0.4rem;
}

.css-1 {
  fill: #CA2055;
  width: 18px;
  height: 18px;
  cursor: pointer;
}

<body>
    <div>
      <div
        class="css-0"
      >
        <svg
          class="css-1"
          color="#CA2055"
          height="24"
          sm="true"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2c5.514 0 10 4.486 10 10s-4.486 10-10 10v-20zm0-2c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12z"
          />
        </svg>
        <p>
          Theme
        </p>
      </div>
    </div>
  </body>,
  "container": .css-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  margin-left: 0.7rem;
  margin-bottom: 1rem;
  cursor: pointer;
}

.css-0 p {
  margin-left: 0.4rem;
}

.css-1 {
  fill: #CA2055;
  width: 18px;
  height: 18px;
  cursor: pointer;
}

<div>
    <div
      class="css-0"
    >
      <svg
        class="css-1"
        color="#CA2055"
        height="24"
        sm="true"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2c5.514 0 10 4.486 10 10s-4.486 10-10 10v-20zm0-2c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12z"
        />
      </svg>
      <p>
        Theme
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
